metadata/src/test/java/org/apache/kafka/image/TopicsImageTest.java

```diff
@@ -176,22 +200,30 @@ private PartitionRegistration newPartition(int[] replicas) {
     public void testBasicLocalChanges() {
         int localId = 3;
         /* Changes already include in DELTA1_RECORDS and IMAGE1:
-         * foo - topic id deleted
+         * foo - topic id deleted then recreated with different topic id
          * bar-0 - stay as follower with different partition epoch
          * baz-0 - new topic to leader
+         * bam - topic id created then deleted
          */
         List<ApiMessageAndVersion> topicRecords = new ArrayList<>(DELTA1_RECORDS);
 
-        // Create a new foo topic with a different id
-        Uuid newFooId = Uuid.fromString("b66ybsWIQoygs01vdjH07A");
+        // Attempt to create a new foo topic with a different id and replica placement
+        topicRecords.add(
+            new ApiMessageAndVersion(
+                new TopicRecord().setName("foo").setTopicId(FOO_UUID3),
+                TOPIC_RECORD.highestSupportedVersion()
+            )
+        );
+        topicRecords.add(newPartitionRecord(FOO_UUID3, 0, Arrays.asList(0, 1, localId)));
```

Comment:
Could you explain this?  We already have a record creating a topic `foo` with `FOO_UUID2`.  I would expect this new record to not be generated by the controller since a topic with that name already exists.  So while I agree that `TopicsDelta` currently will replay it successfully, I'm not sure we should be testing what happens when it does since we never expect this to happen.  In fact, it could be argued that the delta should not replay it.  WDYT?

Reply:
I had a feeling someone might comment on this 😅 I agree we should never be able to get to this point (though arguably we might now with the metadata recovery tool), my intent was to show with this test that we don't do any validation - if that changed in the future, then the test would be changed as well.
I see how it's confusing though, maybe best to remove it.